tools/inundate_mosaic_wrapper.py

```diff
@@ -0,0 +1,143 @@
+        num_workers (int): Number of parallel jobs to run.
+        keep_intermediate (bool): Option to keep intermediate files. Not tested.
+        verbose (bool): Print verbose messages to screen. Not tested.
+        
+    """
+    
+    # Check that output directory exists. Notify user that output directory will be created if not.
+    for output_file in [inundation_raster, inundation_polygon, depths_raster]:
+        if output_file == None:
+            continue
+        parent_dir = os.path.split(output_file)[0]
+        if not os.path.exists(parent_dir):
+            print("Parent directory for " + os.path.split(output_file)[1] + " does not exist. The parent directory will be produced.")
+            os.mkdir(parent_dir)
+    
+    # Check that hydrofabric_dir exists
+    if not os.path.exists(hydrofabric_dir):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), hydrofabric_dir)
+        
+    # Check that huc folder exists in the hydrofabric_dir.
+    if not os.path.exists(os.path.join(hydrofabric_dir, huc)):
+        raise FileNotFoundError((errno.ENOENT, os.strerror(errno.ENOENT), os.path.join(hydrofabric_dir, huc)))
+        
+    # Check that flow file exists
+    if not os.path.exists(flow_file):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), flow_file)
+    
+    # Check job numbers and raise error if necessary
+    total_cpus_available = os.cpu_count() - 1
+    if num_workers > total_cpus_available:
+        raise ValueError('The number of workers (-w), {}, '\
+                          'exceeds your machine\'s available CPU count minus one ({}). '\
+                          'Please lower the num_workers.'.format(num_workers, total_cpus_available))
+    
+    huc_dir = os.path.join(hydrofabric_dir, huc)
+    print("Running inundate for " + huc + "...")
+    # Call Inundate_gms
+    map_file = Inundate_gms(  hydrofabric_dir = hydrofabric_dir, 
+                                     forecast = flow_file, 
+                                     num_workers = num_workers,
+                                     hucs = huc,
+                                     inundation_raster = inundation_raster,
+                                     inundation_polygon = None,
+                                     depths_raster = depths_raster,
+                                     verbose = verbose,
+                                     log_file = None,
+                                     output_fileNames = None )
+
+    print("Mosaicking for " + huc + "...")
+    # Call Mosaic_inundation
+    Mosaic_inundation( map_file,
+                        mosaic_attribute = 'inundation_rasters',
+                        mosaic_output = inundation_raster,
+                        mask = os.path.join(huc_dir,'wbd.gpkg'),
+                        unit_attribute_name = 'huc8',
+                        nodata = -9999,
+                        workers = 1,
+                        remove_inputs = False,
```

Comment:
This can be changed to `remove_inputs = not keep_intermediate` in order to add the ability to keep only the mosaicked raster.